File: javascript/easy/findAnagramMappings.js
/**
 * @param {number[]} A
 * @param {number[]} B
 * @return {number[]}
 */
var anagramMappings = function(A, B) {
    const map = new Map(), res = new Array();
    for(let i = 0; i < B.length; i++) {
        map.set(B[i], i);
    }
    for(let i = 0; i < A.length; i++) {
        res.push(map.get(A[i]));
    }
    return res;
};

File: ruby/easy/find_anagram_mappings.rb
# @param {Integer[]} a
# @param {Integer[]} b
# @return {Integer[]}
def anagram_mappings(a, b)
    map = {}
    res = []
    b.each_with_index do |n, i|
        map[n] = i
    end
    a.each do |n|
        res.push(map[n])
    end
    return res
end
